src/transport/raw/MessageHeader.cpp

```diff
@@ -266,9 +278,13 @@ CHIP_ERROR PayloadHeader::Encode(uint8_t * data, size_t size, uint16_t * encode_
 {
     CHIP_ERROR err = CHIP_NO_ERROR;
     uint8_t * p    = data;
+    uint8_t header = 0;
+
     VerifyOrExit(size >= EncodeSizeBytes(), err = CHIP_ERROR_INVALID_ARGUMENT);
 
-    Write8(p, mExchangeHeader);
+    header = header | mExchangeFlags.Raw();
```

Comment:
why not uint8_t header = mExchangeFlags.Raw()

or even

`Write8(p, mExchangeFlags.Raw())`

?

Are there separate data bits in there? Should we even call it 'header' or should it be exchange flags?

Reply:
To align with  PacketHeader::Encode(), we don't need to assume the size of the mExchangeFlags.Raw() with this way.

Reply:
Acked